qbraid_qir/squin/visitor.py

```diff
@@ -0,0 +1,400 @@
+    measurements: list[int] = field(default_factory=list, init=False)
+
+    def get_num_qubits(self, module):
+        """
+        Extract the number of qubits from the pyqir module.
+
+        Args:
+            module: PyQIR module object
+
+        Returns:
+            int: Number of qubits
+        """
+        # Find the entry point function
+        entry_func = next(filter(pyqir.is_entry_point, module.functions))
+        if not entry_func:
+            return 0
+
+        # Get the required number of qubits and results (classical bits)
+        qubits = pyqir.required_num_qubits(entry_func)
+
+        return qubits
+
+    def lower_qubit_getindex(self, state: lowering.State[pyqir.Module], qid: int):
+        index = qid
+        index_ssa = state.current_frame.push(py.Constant(index)).result
+        qbit_getitem = state.current_frame.push(py.GetItem(self.qreg, index_ssa))
+        return qbit_getitem.result
+
+    def lower_qubit_getindices(self, state: lowering.State[pyqir.Module], qids: list[int]):
+        qbits_getitem = [self.lower_qubit_getindex(state, qid) for qid in qids]
+        qbits = state.current_frame.push(ilist.New(values=qbits_getitem))
+        return qbits.result
+
+    def lower_literal(self, state: lowering.State[pyqir.Module], value) -> ir.SSAValue:
+        raise lowering.BuildError("Literals not supported in pyqir module")
+
+    def lower_global(
+        self, state: lowering.State[pyqir.Module], node: pyqir.Module
+    ) -> lowering.LoweringABC.Result:
+        raise lowering.BuildError("Globals not supported in pyqir module")
+
+    def run(
+        self,
+        module: pyqir.Module,
+        *,
+        globals: dict[str, Any] | None = None,
+        file: str | None = None,
+        lineno_offset: int = 0,
+        col_offset: int = 0,
+        compactify: bool = True,
+        register_as_argument: bool = False,
+        register_argument_name: str = "q",
+    ) -> ir.Region:
+
+        state = lowering.State(
+            self,
+            file=file,
+            lineno_offset=lineno_offset,
+            col_offset=col_offset,
+        )
+
+        with state.frame([module], globals=globals, finalize_next=False) as frame:
+
+            # NOTE: need a register of qubits before lowering statements
+            if register_as_argument:
+                # NOTE: register as argument to the kernel; we have freedom of choice for the name here
+                frame.curr_block.args.append_from(
+                    ilist.IListType[qubit.QubitType, types.Any],
+                    name=register_argument_name,
+                )
+                self.qreg = frame.curr_block.args[0]
+            else:
+                # NOTE: create a new register of appropriate size
+                self.num_qubits = self.get_num_qubits(module)
+                n = frame.push(py.Constant(self.num_qubits))
+                self.qreg = frame.push(func.Invoke((n.result,), callee=qalloc, kwargs=())).result
```

Comment:
I think there's a bug in this line as it breaks when I try to run the example notebook - 

```python
...
--> [270](https://file+.vscode-resource.vscode-cdn.net/Users/thegupta/Desktop/qBraid/repos/qbraid-qir/qbraid_qir/squin/~/Desktop/qBraid/repos/qbraid-qir/qbraid_qir/squin/visitor.py:270)     self.qreg = frame.push(func.Invoke((n.result,), callee=qalloc, kwargs=())).result
    272 self.visit_block(state, module)
    274 if compactify:

TypeError: Invoke.__init__() got an unexpected keyword argument 'kwargs'
```
```suggestion
                self.qreg = frame.push(func.Invoke((n.result,), callee=qalloc)).result
```

Removing the `kwargs` fixes it for me